File: src/main/java/org/openapitools/server/model/OrderTrackDTO.java
package org.openapitools.server.model;

import jakarta.validation.constraints.*;
import jakarta.validation.Valid;


/**
 * Информация о трек-номере посылки (DBS).
 */
public class OrderTrackDTO   {

    private String trackCode;
    private Long deliveryServiceId;

    /**
     * Default constructor.
     */
    public OrderTrackDTO() {
    // JSON-B / Jackson
    }

    /**
     * Create OrderTrackDTO.
     *
     * @param trackCode Трек‑номер посылки.
     * @param deliveryServiceId Идентификатор службы доставки. Информацию о службе доставки можно получить с помощью запроса [GET delivery/services](../../reference/orders/getDeliveryServices.md).
     */
    public OrderTrackDTO(
        String trackCode, 
        Long deliveryServiceId
    ) {
        this.trackCode = trackCode;
        this.deliveryServiceId = deliveryServiceId;
    }



    /**
     * Трек‑номер посылки.
     * @return trackCode
     */
    public String getTrackCode() {
        return trackCode;
    }

    public void setTrackCode(String trackCode) {
        this.trackCode = trackCode;
    }

    /**
     * Идентификатор службы доставки. Информацию о службе доставки можно получить с помощью запроса [GET delivery/services](../../reference/orders/getDeliveryServices.md).
     * @return deliveryServiceId
     */
    public Long getDeliveryServiceId() {
        return deliveryServiceId;
    }

    public void setDeliveryServiceId(Long deliveryServiceId) {
        this.deliveryServiceId = deliveryServiceId;
    }

    /**
      * Create a string representation of this pojo.
    **/
    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        sb.append("class OrderTrackDTO {\n");
        
        sb.append("    trackCode: ").append(toIndentedString(trackCode)).append("\n");
        sb.append("    deliveryServiceId: ").append(toIndentedString(deliveryServiceId)).append("\n");
        sb.append("}");
        return sb.toString();
    }

    /**
     * Convert the given object to string with each line indented by 4 spaces
     * (except the first line).
    */
    private static String toIndentedString(Object o) {
        if (o == null) {
          return "null";
        }
        return o.toString().replace("\n", "\n    ");
    }
}

File: lib/Models/OrderTrackDTO.cpp


#include "OrderTrackDTO.h"

using namespace Tiny;

OrderTrackDTO::OrderTrackDTO()
{
	trackCode = std::string();
	deliveryServiceId = long(0);
}

OrderTrackDTO::OrderTrackDTO(std::string jsonString)
{
	this->fromJson(jsonString);
}

OrderTrackDTO::~OrderTrackDTO()
{

}

void
OrderTrackDTO::fromJson(std::string jsonObj)
{
    bourne::json object = bourne::json::parse(jsonObj);

    const char *trackCodeKey = "trackCode";

    if(object.has_key(trackCodeKey))
    {
        bourne::json value = object[trackCodeKey];



        jsonToValue(&trackCode, value, "std::string");


    }

    const char *deliveryServiceIdKey = "deliveryServiceId";

    if(object.has_key(deliveryServiceIdKey))
    {
        bourne::json value = object[deliveryServiceIdKey];



        jsonToValue(&deliveryServiceId, value, "long");


    }


}

bourne::json
OrderTrackDTO::toJson()
{
    bourne::json object = bourne::json::object();





    object["trackCode"] = getTrackCode();






    object["deliveryServiceId"] = getDeliveryServiceId();



    return object;

}

std::string
OrderTrackDTO::getTrackCode()
{
	return trackCode;
}

void
OrderTrackDTO::setTrackCode(std::string  trackCode)
{
	this->trackCode = trackCode;
}

long
OrderTrackDTO::getDeliveryServiceId()
{
	return deliveryServiceId;
}

void
OrderTrackDTO::setDeliveryServiceId(long  deliveryServiceId)
{
	this->deliveryServiceId = deliveryServiceId;
}



